backend/.gitignore

```diff
@@ -1,6 +1,7 @@
 # compiled output
 /dist
 /node_modules
+package-lock.json
```

Comment:
the package-lock.json file is used by `npm ci` to install the package with the versions specified in the lockfile. Without a lockfile it doesn't work. Is there a specific reason you would not want the package-lock file in the repo?

Also, gitignore rules only affect untracked files. Adding an already tracked file to gitignore does not work, you'll have to untrack the file in question for the gitignore rule to take effect.